src/doc/en/developer/portability_testing.rst

```diff
@@ -949,49 +949,74 @@ Automatic testing on multiple platforms on GitHub Actions
-<https://github.com/sagemath/sage/tree/develop/.github/workflows/tox.yml>`_.
-
-An additional GitHub Actions workflow for testing on Cygwin, not based
-on tox, is defined in the files
-`$SAGE_ROOT/.github/workflows/ci-cygwin*.yml
-<https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-cygwin-standard.yml>`_.
+Actions that runs our portability tests on a multitude of platforms on
+every push of a tag (but not of a branch) to a repository for which
+GitHub Actions are enabled.
+
+In particular, it automatically runs on our main repository sagemath/sage
+on every release tag.
+
+This is defined in the files
+- `$SAGE_ROOT/.github/workflows/ci-linux.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-linux.yml>`_
+  (which calls `$SAGE_ROOT/.github/workflows/docker.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/docker.yml>`_),
+- `$SAGE_ROOT/.github/workflows/ci-macos.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-macos.yml>`_, and
+- `$SAGE_ROOT/.github/workflows/ci-cygwin-standard.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/ci-cygwin-standard.yml>`_
+  (which calls `$SAGE_ROOT/.github/workflows/cygwin.yml
+  <https://github.com/sagemath/sage/tree/develop/.github/workflows/cygwin.yml>`_).
 
 GitHub Actions runs these build jobs on 2-core machines with 7 GB of
 RAM memory and 14 GB of SSD disk space, cf.
 `here <https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources>`_,
-and has a time limit of 6h per job. This is just barely enough for a
-typical ``minimal`` build followed by ``make ptest`` to succeed; and
-plenty of time for a typical ``standard`` build to succeed.
+and has a time limit of 6h per job. This could be just barely enough for a
+typical ``minimal`` build followed by ``make ptest`` to succeed; for
+added robustness, we split it into two jobs. Our workflow stores
+Docker images corresponding to various build phases within these two
+jobs on `GitHub Packages <https://github.com/features/packages>`_ (ghcr.io).
+
+Build logs can be inspected during the run and become available as
+"artifacts" when all jobs of the workflow have finished.  Each job
+generates one tarball.  "Annotations" highlight certain top-level
+errors or warnings issued during the build.
+
+In addition to these automatic runs in our main repository, all Sage
+developers can run the same tests on GitHub Actions in their personal
+forks of the Sage repository. To prepare this, use the following steps:
+
+- `Enable GitHub Actions
+  <https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/enabling-features-for-your-repository/managing-github-actions-settings-for-a-repository#managing-github-actions-permissions-for-your-repository>`_
+  in your fork of the Sage repository.
+
+- Given read/write privileges for packages to GitHub Actions in your
```

Comment:
I don't think it's a good idea to recommend changing the default privileges. Instead, the workflow should specify the required `permissions` it needs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#permissions

Reply:
Thanks, that's probably a good idea.

Reply:
Done now.